transformations/transformations/single_column_coalesced_vector.py

```diff
@@ -76,12 +76,11 @@ def extract_vector_sections(cls, section, horizontal):
         separator_nodes = []
         calls = FindNodes(ir.CallStatement).visit(section)
         for call in calls:
-            if is_loki_pragma(call.pragma, starts_with='inline'):
-                # Because extract_vector_sections was called within pragmas_attached, the pragma node
-                # will not be a part of the returned subsections. Therefore we remove it here to prevent
-                # problems later on
-                call._update(pragma=None)
-                continue
+
+            # check if calls have been enriched
+            if not call.routine is BasicType.DEFERRED:
+                if not horizontal.size in call.routine.symbol_map:
```

Comment:
Uhhh, that's nifty, but could also be quite error-prone. If I understand correctly, if the callee routine deviates from the coding norms, it will now be treated as effectively sequential/non-vector? That could be dangerous as that actually happens quite often in unsanitised code. 

I had originally thought, looking explicitly for the `!$loki routine seq` pragma on the callee (which we already use to bypass SCC-vectorisation), would be more appropriate here (although not quite the one-liner that this is, admittedly)? Would be keen to hear counter-points if I'm not seeing something here? @awnawab

Reply:
Ah yes, fully agreed, if we already have one mechanism to mark routines as sequential, then we should just use that. Thanks for the suggestion, I'll implement that! And also add more tests to boost coverage 😆

Reply:
Also, paging Dr @reuterbal for input on this one?

Reply:
Yeah, I might have to come back to that one as I do have quite a few routines that are unsanitised in that sense. I agree that checking only for horizontal size may be too error-prone, though, so sticking to the existing annotation mechanism would probably be better.